addon/instance-initializers/load-feature-controls.js

```diff
@@ -1,21 +1,21 @@
-import { getStorage } from "ember-local-storage/helpers/storage"
 import config from "ember-get-config"
 
 const { featureControls } = config
 
 export function initialize(appInstance) {
   const features = appInstance.lookup("service:features")
   if (featureControls && featureControls.useLocalStorage) {
-    let featureControlsJSON = getStorage("local")["storage:feature-controls"]
-    if (featureControlsJSON) {
-      const flags = JSON.parse(featureControlsJSON)
-      if (flags) {
-        Object.keys(flags).forEach(flag => {
-          if (features.get("flags").includes(flag)) {
-            flags[flag] ? features.enable(flag) : features.disable(flag)
-          }
-        })
-      }
+
+    const controlStorageService = appInstance.lookup("service:feature-control-storage")
+    // result of controlStorageService.get('featuresLS') is an ObjectProxy we need to use "content"
+    let { content: featureControls } = controlStorageService.get('featuresLS')
```

Comment:
I am surprised, you need to destruct `content` otherwise it does not work?

Because I compare this `featuresLS` (which is a `ObjectProxy` as you say) to the Service [`embedded` from our add-on `ember-cli-embedded`](https://github.com/peopledoc/ember-cli-embedded/blob/8367845de9be87f269ee4057fd6fee378a8a69b2/addon/services/embedded.ts) which also extends `ObjectProxy`. (for the record on RPA HR [we typed it like this](https://github.com/peopledoc/synapps-hr-admin-front/blob/e35eac8f6ab17a27b138ccc11916963b399272e5/types/peopledoc/ember-cli-embedded/index.d.ts))

And all the Tribe has been using this Service such as `this.embedded.get('something')` for instance:

- in [`/app/adapters/application.ts`](https://github.com/peopledoc/synapps-hr-admin-front/blob/e35eac8f6ab17a27b138ccc11916963b399272e5/app/adapters/application.ts)

```ts
export default class ApplicationAdapter extends PDAdapter {
  @service
  declare embedded: IServiceEmbedded

  get host(): string | undefined {
    return this.embedded.get('host')
  }

  get namespace(): string | undefined {
    return this.embedded.get('namespace')
  }
}
```

- in [`/app/router.js`](https://github.com/peopledoc/synapps-hr-admin-front/blob/e35eac8f6ab17a27b138ccc11916963b399272e5/app/router.js)

```js
export default class Router extends EmberRouter {
  @service embedded

  location = config.locationType

  // Use the `rootURL` provided by the host, when the app is started (see `index.html`)
  get rootURL() {
    return this.embedded.get('rootURL')
  }
}
```